WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/CardReaderSettingsKnownViewModel.swift

```diff
@@ -0,0 +1,228 @@
+
+        /// This viewmodel and view supports one reader at a time discovery only.
+        /// TODO: Add another viewmodel and view for listing discovered readers.
+        guard let cardReader = cardReaders.first else {
+            return
+        }
+
+        foundReader = cardReader
+
+        /// If this reader ID is not known to us, we should not connect automatically but first ask the user
+        guard let knownReaderIDs = knownReaderIDs, knownReaderIDs.contains(cardReader.id) else {
+            viewModelState = .foundUnknownReader
+            return
+        }
+
+        /// On the other hand, known readers should be connected to automatically
+        connectToReader()
+    }
+
+    /// Dispatch a request to cancel reader discovery
+    ///
+    func cancelReaderDiscovery() {
+        self.viewModelState = .stoppingSearch
+        cancelReaderDiscovery(completion: { [weak self] in
+            self?.viewModelState = .idle
+        })
+    }
+
+    /// Dispatch a request to connect to the found reader
+    ///
+    func connectToReader() {
+        guard let foundReader = foundReader else {
+            DDLogError("foundReader unexpectedly nil in connectToReader")
+            return
+        }
+
+        viewModelState = .connectingToReader
+
+        ServiceLocator.analytics.track(.cardReaderConnectionTapped)
+        let action = CardPresentPaymentAction.connect(reader: foundReader) { [weak self] result in
+            switch result {
+            case .success(let reader):
+                self?.viewModelState = .connected
+                self?.knownReadersProvider?.rememberCardReader(cardReaderID: reader.id)
+                // If the reader does not have a battery, or the battery level is unknown, it will be nil
+                let properties = reader.batteryLevel
+                    .map { ["battery_level": $0] }
+                ServiceLocator.analytics.track(.cardReaderConnectionSuccess, withProperties: properties)
+            case .failure(let error):
+                self?.viewModelState = .connectionFailure(error)
+                ServiceLocator.analytics.track(.cardReaderConnectionFailed, withError: error)
+            }
+        }
+        ServiceLocator.stores.dispatch(action)
+    }
+
+    /// Discard the found reader and keep searching
+    /// As discussed in p91TBi-5fB-p2#comment-4849, for the first release,
+    /// we will restart the discovery process again
+    func continueSearch() {
+        foundReader = nil
+        cancelReaderDiscovery { [weak self] in
+            self?.startReaderDiscovery()
+        }
+    }
+
+    /// Updates whether the view this viewModel is associated with should be shown or not
+    /// Notifes the viewModel owner if a change occurs via didChangeShouldShow
+    ///
+    private func reevaluateShouldShow() {
+        var newShouldShow: CardReaderSettingsTriState = .isUnknown
+
+        if ( noKnownReaders == .isUnknown ) || ( noConnectedReaders == .isUnknown ) {
```

Comment:
I'm surprised the linter didn't yell at you because of the spaces after the parenthesis 🤷🏼

Reply:
Huh! Me too. Other coding standards die hard.

Reply:
Fixed in https://github.com/woocommerce/woocommerce-ios/pull/4565/commits/82ca59bdac562dc1a7f3408ff317d37f20fbabfc